File: java/src/main/java/com/douglasmatosdev/service/BitcoinPriceLogger.java
package com.douglasmatosdev.service;

public class BitcoinPriceLogger implements BitcoinPriceObserver {

    @Override
    public void update(Double oldPrice, Double newPrice) {
        // Here we can implement the logic to log the price of Bitcoin
        System.out.println("\nBitcoinPriceLogger");
        System.out.println("Price updated: " + oldPrice + " -> " + newPrice);
    }
}

File: typescript/services/BitcoinPriceLogger.ts
import { BitcoinPriceObserver } from "../services/BitcoinPriceObserver";

export class BitcoinPriceLogger implements BitcoinPriceObserver {
    update(oldPrice: number, newPrice: number): void {
        console.log('\n\nBitcoinPriceLogger');
        console.log(`Price updated: ${oldPrice} -> ${newPrice}`);
    }
}